registry/machine.go

```diff
@@ -93,9 +106,78 @@ func (r *EtcdRegistry) SetMachineState(ms machine.MachineState, ttl time.Duratio
 		return uint64(0), err
 	}
 
+	// Set the initial metadata on creation
+	if err = r.setMachineMetadata(ms); err != nil {
+		return uint64(0), err
+	}
+
 	return resp.Node.ModifiedIndex, nil
 }
 
+func (r *EtcdRegistry) UpdateMachineMetadata(machID string, key string, value string) error {
+	//Attempt to update key
+	update := etcd.Update{
```

Comment:
This was patterned after SetMachineState above. I agree Set makes more sense and have modified the code to use it, should SetMachineState be modified as well?